cmd/kube-apiserver/app/server.go

```diff
@@ -510,13 +514,26 @@ func getRuntimeConfigValue(s *options.APIServer, apiKey string, defaultValue boo
 	return defaultValue
 }
 
-// Parses the given runtime-config and formats it into map[string]ApiGroupVersionOverride
-func parseRuntimeConfig(s *options.APIServer) (map[string]genericapiserver.APIGroupVersionOverride, error) {
+// Parses the given runtime-config and formats it into genericapiserver.APIResourceEnablement
+func parseRuntimeConfig(s *options.APIServer) (genericapiserver.APIResourceEnablement, error) {
+	v1GroupVersionString := "api/v1"
+	extensionsGroupVersionString := "extensions/v1beta1"
+	batchGroupVersionString := "batch/v1"
+	autoscalingGroupVersionString := "autoscaling/v1"
+	versionToEnablementString := map[unversioned.GroupVersion]string{
```

Comment:
> Why not just make your new type directly parse a runtime config string? Why do we need this map?

Because this logic is pretty convoluted.  I'd like to keep the default impl something reasonable and something that someone else downstream might actually use.

This mismash of default on, default off, some combinations override, some combinations don't is backwards compatible with this file, but it isn't a sane object to create.